src/features/goblins/trader/tradingPost/Trader.tsx

```diff
@@ -16,11 +17,27 @@ export const Trader: React.FC = () => {
 
   const [showModal, setShowModal] = useState(false);
 
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  // Support deep-link to auto-open the trader
+  useEffect(() => {
+    if (!searchParams.has("viewListingsForLand")) return;
+    if (!goblinState.matches("playing")) return;
+    openTrader();
+  }, [goblinState, searchParams]);
+
   const openTrader = () => {
     goblinService.send("OPEN_TRADING_POST");
     setShowModal(true);
   };
 
+  const handleModalClose = () => {
+    const modifiedParams = Object.assign({}, searchParams);
```

Comment:
```suggestion
    const modifiedParams = new URLSearchParams(searchParams);
```

Awesome PR @ShiningMinions ! 🎉 

It's working really nicely. I did encounter one small issue testing. When closing the trader modal, I got the error below on Firefox:

`Uncaught TypeError: modifiedParams.delete is not a function`.

Replacing `modifiedParams` with `const modifiedParams = new URLSearchParams(searchParams);
` seems to resolve this issue.